src/main/java/com/github/manolo8/darkbot/core/objects/facades/SlotBarsProxy.java

```diff
@@ -12,9 +13,22 @@
     //public SlotBar premiumBar    = new SlotBar(); //104
     //public SlotBar proActionBar  = new SlotBar(); //112
 
+    private Main main;
+
     @Override
     public void update() {
         this.categoryBar.update(API.readMemoryLong(address + 88));
         this.standardBar.update(API.readMemoryLong(address + 96));
     }
+
+    public String getItemIdByKey(Character c) {
+        SettingsProxy.KeyBind keybind = main.facadeManager.settings.getAtChar(c);
```

Comment:
This will throw a NPE.

Reply:
What to do? I used `if (keybind == null) return null;` below

Reply:
read the comment above. `main` will be null

Reply:
How can i initialize, i have no idea about that. inside update() ?